OData/src/System.Web.OData/OData/Query/Expressions/SelectExpandBinder.cs

```diff
@@ -391,7 +391,7 @@ private Expression CreateTotalCountExpression(Expression source, ExpandedNavigat
                 Expression propertyName = CreatePropertyNameExpression(elementType, propertyToExpand, source);
                 Expression propertyValue = CreatePropertyValueExpressionWithFilter(elementType, propertyToExpand, source,
                     expandItem.FilterOption);
-                Expression nullCheck = Expression.Equal(propertyValue, Expression.Constant(null));
+                Expression nullCheck = GetNullCheckExpression(propertyToExpand, propertyValue, projection);
```

Comment:
@VikingsFan `IsNull` property presence does not depend on the value of HandleNullPropagation.
When `HandleNullPropagation == HandleNullPropagationOption.False` then 
```C#
IsNull = (Convert(Param_1.Customer.ID) == null)
```
When `HandleNullPropagation == HandleNullPropagationOption.True` then 
```C#
IsNull = (IIF((IIF((Param_1 == null), null, Param_1.Customer) == null), null, Convert(IIF((Param_1 == null), null, Param_1.Customer).ID)) == null)
```